ethdb/database.go

```diff
@@ -129,9 +154,11 @@ type AncientWriter interface {
 // AncientWriteOp is given to the function argument of ModifyAncients.
 type AncientWriteOp interface {
 	// Append adds an RLP-encoded item.
+	// Returns ErrAncientOutOrderInsertion if attempting to insert items out of order.
```

Comment:
I think this kind should also be checked and return the `ErrAncientUnknownKind` if necessary?

Reply:
Hmmm. `freezerBatch` currently panic if an unknown kind is used. While that would make sense to return `ErrAncientUnknownKind`, it also would be a programming error to provide an unknown kind rather than a runtime problem, so a panic seems more appropriate.
Which one do you prefer?